src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -5172,6 +5182,25 @@ instruct storeNKlass(iRegN src, memory mem)
   ins_pipe(istore_reg_mem);
 %}
 
+// ============================================================================
+// Prefetch instructions
+// Must be safe to execute with invalid address (cannot fault).
+
+instruct prefetchalloc( memory mem ) %{
+  match(PrefetchAllocation mem);
+
+  ins_cost(ALU_COST * 3);
+  format %{ "prefetch_w $mem\t# Prefetch for write" %}
+
+  ins_encode %{
+    __ addi(t0, as_Register($mem$$base), $mem$$disp);
+    __ andi(t0, t0, ~(CacheLineSize - 1));
```

Comment:
Do we really need to align to CacheLineSize here and in generate_prefetch?
I didn't see this restriction from the official CMO specification [1] about this requirement.
Could you please confirm that? Thanks.

[1] https://github.com/riscv/riscv-CMOs/blob/master/specifications/cmobase-v1.0.1.pdf

Reply:
It's indeed not a requirement, let me remove that `andi`.